docs/debugging-testing/testing/unit_testing.md

```diff
@@ -124,76 +135,40 @@ With the following steps, you can write a unit test. This example creates dummy
     #endif
     ```
 
-    **MyClass.cpp**
+1. Add a new `mbed-headers-example` interface library: 
 
-    ```
-    #include "MyClass.h"
-    #include "OtherClass.h"
-
-    namespace example {
-
-    int MyClass::myFunction() {
-        OtherClass o = OtherClass();
-        return o.otherFunction();
-    }
-
-    }
-    ```
-
-    **OtherClass.h**
+   ### Stub CMake input source file
 
+   ```
+   target_include_directories(mbed-headers-example
+    INTERFACE
```

Comment:
Why not? If someone asks you to deliver all the hamburgers, why wouldn't you deliver _all_ the hamburgers? If users ask us to build all targets, we should build all targets. Building everything, actually everything, is very useful in CI.

Reply:
There are 2 reasons
1. One to avoid build all target
2. This library holds only header files and no source files

Make the all headers library (`mbed-headers-base, mbed-headers-platform, mbed-headers-connectivity etc.,`) into object library and `INTERFACE` `target_link_libraries` with `mbed-headers`, what is your suggestion

Reply:
If I'm not mistaken, you use EXCLUDE_FROM_ALL to avoid adding a library or subdirectory to the "all" target. INTERFACE just means that the library doesn't compile any sources or produce any output artifacts. 

If these are header-only libraries, INTERFACE is the semantically correct thing here, but it isn't stopping anything being added to the "all" target.

Reply:
Okay, I think we can use INTERFACE as it is header-only libraries and doesn't produce any output artifacts